src/main/reaktivity/protocol.idl

```diff
@@ -0,0 +1,780 @@
+
+        variant AmqpByte switch (AmqpType) of int8
+        {
+            case BYTE: int8;
+        }
+
+        variant AmqpShort switch (AmqpType) of int16
+        {
+            case SHORT: int16;
+        }
+
+        variant AmqpInt switch (AmqpType) of int32
+        {
+            case INT4: int32;
+            case INT1: int8;
+        }
+
+        variant AmqpLong switch (AmqpType) of int64
+        {
+            case LONG8: int64;
+            case LONG1: int8;
+        }
+
+        variant AmqpFloat switch (AmqpType)
+        {
+            case FLOAT: octets[4];
+        }
+
+        variant AmqpDouble switch (AmqpType)
+        {
+            case DOUBLE: octets[8];
+        }
+
+        variant AmqpDecimal32 switch (AmqpType)
+        {
+            case DECIMAL32: octets[4];
+        }
+
+        variant AmqpDecimal64 switch (AmqpType)
+        {
+            case DECIMAL64: octets[8];
+        }
+
+        variant AmqpDecimal128 switch (AmqpType)
+        {
+            case DECIMAL128: octets[16];
+        }
+
+        variant AmqpChar switch (AmqpType)
+        {
+            case CHAR: octets[4];
+        }
+
+        variant AmqpTimestamp switch (AmqpType)
+        {
+            case TIMESTAMP: int64;
+        }
+
+        variant AmqpUUId switch (AmqpType)
+        {
+            case UUID: string16;
+        }
+
+        variant AmqpBinary switch (AmqpType)
+        {
+            case BINARY1: octets[uint8];
+            case BINARY4: octets[uint32];
+        }
+
+        variant AmqpString switch (AmqpType) of string32
+        {
+            case STRING1: string;
+            case STRING4: string32;
+        }
+
+        variant AmqpSymbol switch (AmqpType) of string32
+        {
+            case SYMBOL1: string;
+            case SYMBOL4: string32;
+        }
+
+        variant AmqpList switch (AmqpType) of list
+        {
+            case LIST4: list[uint32, uint32];
+            case LIST1: list[uint8, uint8];
+            case LIST0: list[0, 0];
+        }
+
+        variant AmqpMap switch (AmqpType) of map
+        {
+            case MAP4: map[uint32];
+            case MAP1: map[uint8];
+        }
+
+        variant AmqpArray switch (AmqpType) of array
+        {
+            case ARRAY4: array[uint32, uint32];
+            case ARRAY1: array[uint8, uint8];
+        }
+
+        // 2.7 Performatives
+        list AmqpOpen (AmqpList)
+        {
+            required AmqpString containerId;
+            AmqpString hostname;
+            AmqpUInt maxFrameSize;
+            AmqpUShort channelMax;
+            AmqpMilliseconds idleTimeOut;
+            AmqpArray<AmqpIETFLanguageTag> outgoingLocales;
+            AmqpArray<AmqpIETFLanguageTag> incomingLocales;
+            AmqpArray<AmqpSymbol> offeredCapabilities;
+            AmqpArray<AmqpSymbol> desiredCapabilities;
+            AmqpFields<AmqpValue> properties;
+        }
+
+        list AmqpBegin (AmqpList)
+        {
+            AmqpUShort remoteChannel;
```

Comment:
what does it mean for `remoteChannel` not to be `required` when subsequent fields are `required`?